langchain/document_loaders/trello.py

```diff
@@ -0,0 +1,126 @@
+
+    Args:
+        api_key (str): Trello API key.
+        api_token (str): Trello API token.
+        board_name (str): The name of the Trello board.
+        include_card_name (bool): Whether to include the name of the card in the document. Defaults to False.
+        include_comments (bool): Whether to include the comments on the card in the document. Defaults to False.
+        include_checklist (bool): Whether to include the checklist on the card in the document. Defaults to False.
+        card_filter (str, optional): Use "closed" / "open". Defaults to "all".
+        extra_metadata (tuple[str]): List of additional metadata fields to include as document metadata. Defaults to ["due_date", "labels", "list", "is_closed"].
+
+    """
+
+    def __init__(
+        self, 
+        api_key: str, 
+        api_token: str,
+        board_name: str,
+        include_card_name: bool = True,
+        include_comments: bool = True,
+        include_checklist: bool = True,
+        card_filter: str = "all",
+        extra_metadata: tuple[str, ...] = ("due_date", "labels", "list", "is_closed"),
+    ):
+        """Initialize Trello loader."""
+        self.api_key = api_key
+        self.api_token = api_token
+        self.board_name = board_name
+        self.include_card_name = include_card_name
+        self.include_comments = include_comments
+        self.include_checklist = include_checklist
+        self.extra_metadata = extra_metadata
+        self.card_filter = card_filter
+
+    def load(self) -> List[Document]:
+        """Loads all cards from the specified Trello board.
+        You can filter the cards, metadata and text included by using the optional parameters.
+
+         Returns:
+            A list of documents, one for each card in the board.
+        """
+
+        try:
+            from trello import TrelloClient  # type: ignore
+        except ImportError as ex:
+            raise ImportError(
+                "Could not import trello python package. "
+                "Please install it with `pip install py-trello`."
+            ) from ex
+        try:
+            from bs4 import BeautifulSoup  # type: ignore
+        except ImportError as ex:
+            raise ImportError(
+                "`beautifulsoup4` package not found, please run"
+                " `pip install beautifulsoup4`"
+            ) from ex
+
+        docs: List[Document] = []
+        client = TrelloClient(api_key=self.api_key, token=self.api_token)
+
+        # Find the board with the matching name
+        board = next((b for b in client.list_boards()
+                     if b.name == self.board_name), None)
+        if board:
+            board_id = board.id
+        else:
+            raise ValueError(f"Board `{self.board_name}` not found.")
+
+        # Create a dictionary with the list IDs as keys and the list names as values
+        list_dict = {list_item.id: list_item.name for list_item in board.list_lists()}
```

Comment:
board.list_lists() :
`[
  {
    "id": "5555cacbc4daa90564b34cf2",
    "name": "Publishing Considerations"
  },
  {
    "id": "5555059b74c03b3a9e362cd0",
    "name": "Backlog"
  },
  {
    "id": "555505a3427fd688c1ca5ebd",
    "name": "Selected for Milestone"
  },
  {
    "id": "555505ba95ff925f9fb1b370",
    "name": "Blocked"
  },
  {
    "id": "555505a695ff925f9fb1b13d",
    "name": "In Progress"
  },
  {
    "id": "555505bdfe380c7edc8ca1a3",
    "name": "Done"
  }
]`